core/src/test/scala/org/apache/spark/deploy/master/MasterSuite.scala

```diff
@@ -157,4 +161,27 @@ class MasterSuite extends FunSuite with Matchers {
     CustomRecoveryModeFactory.instantiationAttempts should be > instantiationAttempts
   }
 
+  test("Master & worker web ui available") {
+    implicit val formats = org.json4s.DefaultFormats
+    val conf = new SparkConf()
+    val localCluster = new LocalSparkCluster(2, 2, 512, conf)
+    localCluster.start()
+    try {
+      eventually(timeout(2 seconds), interval(100 milliseconds)) {
```

Comment:
Is 2 seconds enough? Does `start()` guarantee that the WebUI is up and running when it returns?

I'm wary of some really slow build machine triggering this timeout.